configure

```diff
@@ -292,277 +332,749 @@ LIBS="${LIBS} mpfr"
-          echo "****WARNING**** GC does not support TLS....disabling TLS"
-	  echo "GC does not support TLS....disabling TLS" >> config.log
-      fi
-      cp fmpz/link/fmpz_gc.c fmpz/fmpz.c
-      cp fmpz-conversions-gc.in fmpz-conversions.h
+    TLS=0
+    if [ "$WANT_TLS" = "1" ]; then
+        echo "****WARNING**** GC does not support TLS....disabling TLS"
+        echo "GC does not support TLS....disabling TLS" >> config.log
+    fi
+    cp fmpz/link/fmpz_gc.c fmpz/fmpz.c
+    cp fmpz-conversions-gc.in fmpz-conversions.h
 else
-   if [ "$REENTRANT" = "1" ]; then
-      cp fmpz/link/fmpz_reentrant.c fmpz/fmpz.c
-      cp fmpz-conversions-reentrant.in fmpz-conversions.h
-   else
-      cp fmpz/link/fmpz_single.c fmpz/fmpz.c
-      cp fmpz-conversions-single.in fmpz-conversions.h
-   fi
+    if [ "$REENTRANT" = "1" ]; then
+        cp fmpz/link/fmpz_reentrant.c fmpz/fmpz.c
+        cp fmpz-conversions-reentrant.in fmpz-conversions.h
+    else
+        cp fmpz/link/fmpz_single.c fmpz/fmpz.c
+        cp fmpz-conversions-single.in fmpz-conversions.h
+    fi
 fi
-# Architecture handler
 
-KERNEL=`uname`
+################################################################################
+################################################################################
+# build handler
+################################################################################
+################################################################################
+
+BUILD_KERNEL=`uname`
 
+echo "Checking build..."
+echo "Checking build..." >> config.log
 if [ -z "$BUILD" ]; then
-   ARCH=`uname -m`
-
-   if [ "$(uname | cut -d_ -f1)" = "MSYS" ]; then
-      if [ "$ARCH" = "x86_64" ]; then
-         OS="MINGW64"
-      else
-         OS="MINGW32"
-      fi
-   elif [ "$(uname | cut -d_ -f1)" = "MINGW32" ]; then
-      if [ "$ABI" = "64" ]; then
-         OS="MINGW64"
-      else
-         OS="MINGW32"
-      fi 
-   elif [ "$(uname | cut -d_ -f1)" = "MINGW64" ]; then
-      if [ "$ABI" = "32" ]; then
-         OS="MINGW32"
-      else
-         OS="MINGW64"
-      fi
-   elif [ "$(uname | cut -d_ -f1)" = "CYGWIN" ]; then
-      if [ "$ARCH" = "x86_64" ]; then
-         if [ "$ABI" = "32" ]; then
-            OS="CYGWIN32"
-         else
-            OS="CYGWIN64"
-            ABI="64"
-         fi
-      else
-         OS="CYGWIN32"
-      fi
-   else
-      OS=`uname -s`
-   fi
+    # Build was not specified, so try to guess it
+
+    # Set MARCH_FLAG to -march=native -mno-avx512f, where the second option is
+    # to disable AVX512 on those systems that supports it.
+    MARCH_FLAG="-march=native -mno-avx512f"
+
+    ############################################################################
+    # architecture and machine
+    ############################################################################
+    BUILD_ARCH=`uname -m`
+
+    case "$BUILD_ARCH" in
+        x86_64 | amd64)
+            BUILD_MACHINE="x86_64"
+            ;;
+        x86 | i*86 | pc)
```

Comment:
Where are you getting "pc" from? Also, why `i*86`? Do you mean `i[3-6]`86`? Though realistically I doubt it'll work on a 386 :joy:

Reply:
MPIR needed to use pc for some machines. But this could be due to it being based on GMP which has its own custom version of config.guess and config.sub. Unfortunately the FSF ones weren't very useful or portable for an assembly library. So this might explain where this has come from.